File: ios/RnSampleModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RnSampleModule, NSObject)

RCT_EXTERN_METHOD(getDeviceId: (RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/RnSampleModule.swift
import UIKit;

@objc(RnSampleModule)
class RnSampleModule: NSObject {

    @objc(getDeviceId:withRejecter:)
    func getDeviceId(resolve: RCTPromiseResolveBlock,
                     reject:RCTPromiseRejectBlock) -> Void {
        let id = UIDevice.current.identifierForVendor?.uuidString
        resolve(id);
    }
}
